### PR TITLE
Cleans up the forced gravity component and makes it use signals

### PR DESCRIPTION
**Pull request by ninjanomnom**

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -446,27 +446,23 @@ GLOBAL_LIST_EMPTY(teleportlocs)
...
+		return FALSE
+	
+	var/turf_comp_flags = SEND_SIGNAL(src, COMSIG_TURF_HAS_GRAVITY, src)
+	if(turf_comp_flags & COMPONENT_FORCE_GRAVITY_ON)
+		return TRUE
+	if(turf_comp_flags & COMPONENT_FORCE_GRAVITY_OFF)
+		return FALSE
 
 	var/area/A = get_area(T)
 	if(isspaceturf(T)) // Turf never has gravity
```
Comment (vuonojenmustaturska) [inline code/game/area/areas.dm]: move this if check before the `get_area()` above for a tiny perf gain in this path

**Comment by AnturK** [comment]
The whole point of this using get instead of a signal was because the value is a scalar. You're just removing the functionality right now.

**Comment by ninjanomnom** [comment]
Oh I was not aware gravity was anything other than a bool, has_gravity is a really poor name for that now then. Well, that's easy to fix

File: code/datums/components/forced_gravity.dm [hunk 2]
```diff
@@ -1,8 +1,20 @@
 /datum/component/forced_gravity
-	var/gravity = 1
+	var/gravity
 	var/ignore_space = FALSE	//If forced gravity should also work on space turfs
 
-/datum/component/forced_gravity/Initialize(forced_value = 1)
+/datum/component/forced_gravity/Initialize(forced_value = TRUE)
```
Comment (AnturK) [inline code/datums/components/forced_gravity.dm]: TRUE => 1

**Review by AnturK — CHANGES_REQUESTED** [review]
Looks good now. (I still dislike list passing format)

**Review by AnturK — APPROVED** [review]

**Pull request merged**